core/src/main/scala/quasar/destination/gbq/GBQFlow.scala

```diff
@@ -246,11 +250,19 @@ final class GBQFlow[F[_]: Concurrent](
       Resource.eval(jobReq).flatMap(client.run).use { resp =>
         resp.status match {
           case Status.Ok => resp.headers match {
-            case Location(loc) => loc.uri.asRight[InitializationError[Json]].pure[F]
+            case Location(loc) => 
+              Sync[F].delay(log.info(s"Successfully initialised the job under URI '${loc.uri}'"))
+                .as(loc.uri.asRight[InitializationError[Json]])
           }
-          case _ =>  DestinationError.malformedConfiguration(
-            (GBQDestinationModule.destinationType, jString("Reason: " + resp.status.reason),
-            config.sanitizedJson.toString)).asLeft[Uri].pure[F]
+          case otherStatus =>  
+            resp.attemptAs[String].fold(
+                _ => Sync[F].delay(log.error(s"GBQ job creation failed with status '$otherStatus' and no body")),
+                body => Sync[F].delay(log.error(s"GBQ job creationg failed with status '$otherStatus': $body")))
+              .as(DestinationError.malformedConfiguration(
```

Comment:
We use `malformedConfirmation` for when the json format is incorrect, like here: https://github.com/precog/quasar-destination-gbq/blob/3ca1ef7d9fafe8ab9b5503cfe40c192ce3317471/core/src/main/scala/quasar/destination/gbq/GBQDestination.scala#L77

I think this could be an `InvalidConfiguration`. I'm ok with `InvalidConfiguration` for all statuses but optionally this could be an AccessDenied, if that's what the status indicates.

Actually this looks wrong in the same way https://github.com/precog/quasar-destination-gbq/blob/3ca1ef7d9fafe8ab9b5503cfe40c192ce3317471/core/src/main/scala/quasar/destination/gbq/GBQDestination.scala#L105